src/services/motion/types.ts

```diff
@@ -57,3 +59,20 @@ export const encodeConstraints = (obj: Constraints): pb.Constraints => {
 
   return result;
 };
+
+export const encodeMotionConfiguration = (
+  obj: MotionConfiguration
+): pb.MotionConfiguration => {
+  const result = new pb.MotionConfiguration();
+
+  result.setVisionServicesList(
+    obj.visionServicesList.map((x) => encodeResourceName(x))
+  );
+  result.setPositionPollingFrequencyHz(obj.positionPollingFrequencyHz);
+  result.setObstaclePollingFrequencyHz(obj.obstaclePollingFrequencyHz);
+  result.setPlanDeviationM(obj.planDeviationM);
+  result.setLinearMPerSec(obj.linearMPerSec);
+  result.setAngularDegsPerSec(obj.angularDegsPerSec);
+
+  return result;
+};
```

Comment:
annoying, all of these fields are required in the resulting protobuf object, even tho they are marked as `optional` in the [source proto file](https://github.com/viamrobotics/api/blob/037415ecb066649cdbc955e2667902dac57c0fbf/proto/viam/service/motion/v1/motion.proto#L84-L97)

Reply:
That's weird. Why are they required if they're supposed to be optional?

Reply:
yeah I'm not sure... but we can resolve this later since make them optional will still be backwards compatible - for now i just want to get his building again